### ISSUE TITLE
client build fails on LinuxMint-19.1

### ISSUE THREAD

**Issue by sidommi** [Genymobile/scrcpy#1754]
**Environment**
 - OS: [Linux Mint 19.1]
 - scrcpy version: [1.8]
 - installation method: [manual build of client using pre-built server]

I have already installed libsdl2-dev package and then have the following issue. Please help.
I have removed a part of the terminal outpu since it is too long to post. I have attached the full version in a text file.
Thank you ,
Sid

PS:- for some reason I am unable to type set the terminal output as a code. Sorry.

```
(base) shommi2016@PC-GC84:~$ cd scrcpy/x/
(base) shommi2016@PC-GC84:~/scrcpy/x$ ninja
[1/26] Compiling C object app/scrcpy.p/src_controller.c.o
FAILED: app/scrcpy.p/src_controller.c.o 
/home/shommi2016/anaconda3/bin/x86_64-conda_cos6-linux-gnu-cc -Iapp/scrcpy.p -Iapp -I../app -I../app/src -I/usr/include/x86_64-linux-gnu -I/usr/include/SDL2 -flto -fdiagnostics-color=always -pipe -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wextra -std=c11 -O3 -march=nocona -mtune=haswell -ftree-vectorize -fPIC -fstack-protector-strong -fno-plt -O2 -ffunction-sections -isystem /home/shommi2016/anaconda3/include -DNDEBUG -D_FORTIFY_SOURCE=2 -O2 -isystem /home/shommi2016/anaconda3/include -D_REENTRANT -MD -MQ app/scrcpy.p/src_controller.c.o -MF app/scrcpy.p/src_controller.c.o.d -o app/scrcpy.p/src_controller.c.o -c ../app/src/controller.c
In file included from ../app/src/controller.c:1:0:
../app/src/controller.h:5:10: fatal error: SDL2/SDL_mutex.h: No such file or directory
 #include <SDL2/SDL_mutex.h>
          ^~~~~~~~~~~~~~~~~~
compilation terminated.
[2/26] Compiling C object app/scrcpy.p/src_device.c.o
FAILED: app/scrcpy.p/src_device.c.o 
/home/shommi2016/anaconda3/bin/x86_64-conda_cos6-linux-gnu-cc -Iapp/scrcpy.p -Iapp -I../app -I../app/src -I/usr/include/x86_64-linux-gnu -I/usr/include/SDL2 -flto -fdiagnostics-color=always -pipe -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wextra -std=c11 -O3 -march=nocona -mtune=haswell -ftree-vectorize -fPIC -fstack-protector-strong -fno-plt -O2 -ffunction-sections -isystem /home/shommi2016/anaconda3/include -DNDEBUG -D_FORTIFY_SOURCE=2 -O2 -isystem /home/shommi2016/anaconda3/include -D_REENTRANT -MD -MQ app/scrcpy.p/src_device.c.o -MF app/scrcpy.p/src_device.c.o.d -o app/scrcpy.p/src_device.c.o -c ../app/src/device.c
In file included from ../app/src/device.h:6:0,
                 from ../app/src/device.c:1:
app/config.h:20:0: warning: "NDEBUG" redefined
 #define NDEBUG
 
<command-line>:0:0: note: this is the location of the previous definition
In file included from /home/shommi2016/anaconda3/x86_64-conda_cos6-linux-gnu/sysroot/usr/include/features.h:356:0,
                 from /home/shommi2016/anaconda3/x86_64-conda_cos6-linux-gnu/sysroot/usr/include/stdint.h:26,
                 from /home/shommi2016/anaconda3/lib/gcc/x86_64-conda_cos6-linux-gnu/7.3.0/include/stdint.h:9,
                 from ../app/src/common.h:4,
                 from ../app/src/device.h:7,
                 from ../app/src/device.c:1:
/usr/include/x86_64-linux-gnu/sys/cdefs.h:467:49: error: missing binary operator before token "("
 #if __GNUC_PREREQ (4,8) || __glibc_clang_prereq (3,5)
                                                 ^
In file included from ../app/src/device.h:8:0,
                 from ../app/src/device.c:1:
../app/src/util/net.h:6:10: fatal error: SDL2/SDL_platform.h: No such file or directory
 #include <SDL2/SDL_platform.h>
          ^~~~~~~~~~~~~~~~~~~~~
compilation terminated.

.
.
[FIND IN FILE ATTACHED THE FULL ERROR]
.
.

In file included from /usr/include/x86_64-linux-gnu/libavutil/common.h:42:0,
                 from /usr/include/x86_64-linux-gnu/libavutil/avutil.h:296,
                 from /usr/include/x86_64-linux-gnu/libavutil/samplefmt.h:24,
                 from /usr/include/x86_64-linux-gnu/libavcodec/avcodec.h:31,
                 from /usr/include/x86_64-linux-gnu/libavformat/avformat.h:319,
                 from ../app/src/main.c:6:
/usr/include/x86_64-linux-gnu/libavutil/attributes.h:94:34: error: expected declaration specifiers before '__attribute__'
 #    define attribute_deprecated __attribute__((deprecated))
                                  ^
/usr/include/x86_64-linux-gnu/libavformat/avformat.h:2969:1: note: in expansion of macro 'attribute_deprecated'
 attribute_deprecated
 ^~~~~~~~~~~~~~~~~~~~
In file included from ../app/src/main.c:6:0:
/usr/include/x86_64-linux-gnu/libavformat/avformat.h:2974:1: warning: empty declaration
 enum AVTimebaseSource {
 ^~~~
/usr/include/x86_64-linux-gnu/libavformat/avformat.h:2993:57: warning: type defaults to 'int' in declaration of 'AVOutputFormat' [-Wimplicit-int]
 int avformat_transfer_internal_stream_timing_info(const AVOutputFormat *ofmt,
                                                         ^~~~~~~~~~~~~~
/usr/include/x86_64-linux-gnu/libavformat/avformat.h:2993:72: error: expected ';', ',' or ')' before '*' token
 int avformat_transfer_internal_stream_timing_info(const AVOutputFormat *ofmt,
                                                                        ^
/usr/include/x86_64-linux-gnu/libavformat/avformat.h:3002:1: error: expected declaration specifiers before 'AVRational'
 AVRational av_stream_get_codec_timebase(const AVStream *st);
 ^~~~~~~~~~
../app/src/main.c:8:10: fatal error: SDL2/SDL.h: No such file or directory
 #include <SDL2/SDL.h>
          ^~~~~~~~~~~~
compilation terminated.
ninja: build stopped: subcommand failed.
```
[scrcpy_build_error_full.txt](https://github.com/Genymobile/scrcpy/files/5208537/scrcpy_build_error_full.txt)

**Comment by rom1v** [comment]
> scrcpy version: [1.8]

Why 1.8?

Remove the `x` directory, and execute `meson x`. What is the output?

**Comment by sidommi** [comment]
1.8 because the prebuilt server .jar was that;

the following output for your instructions.

```
(base) shommi2016@PC-GC84:~/scrcpy$ meson x
The Meson build system
Version: 0.55.1
Source dir: /home/shommi2016/scrcpy
Build dir: /home/shommi2016/scrcpy/x
Build type: native build
Project name: scrcpy
Project version: 1.16
Using 'CC' from environment with value: '/home/shommi2016/anaconda3/bin/x86_64-conda_cos6-linux-gnu-cc'
Using 'CFLAGS' from environment with value: '-march=nocona -mtune=haswell -ftree-vectorize -fPIC -fstack-protector-strong -fno-plt -O2 -ffunction-sections -pipe -isystem /home/shommi2016/anaconda3/include'
Using 'LDFLAGS' from environment with value: '-Wl,-O2 -Wl,--sort-common -Wl,--as-needed -Wl,-z,relro -Wl,-z,now -Wl,--disable-new-dtags -Wl,--gc-sections -Wl,-rpath,/home/shommi2016/anaconda3/lib -Wl,-rpath-link,/home/shommi2016/anaconda3/lib -L/home/shommi2016/anaconda3/lib'
Using 'CPPFLAGS' from environment with value: '-DNDEBUG -D_FORTIFY_SOURCE=2 -O2 -isystem /home/shommi2016/anaconda3/include'
Using 'AR' from environment with value: '/home/shommi2016/anaconda3/bin/x86_64-conda_cos6-linux-gnu-ar'
Using 'CC' from environment with value: '/home/shommi2016/anaconda3/bin/x86_64-conda_cos6-linux-gnu-cc'
Using 'CFLAGS' from environment with value: '-march=nocona -mtune=haswell -ftree-vectorize -fPIC -fstack-protector-strong -fno-plt -O2 -ffunction-sections -pipe -isystem /home/shommi2016/anaconda3/include'
Using 'LDFLAGS' from environment with value: '-Wl,-O2 -Wl,--sort-common -Wl,--as-needed -Wl,-z,relro -Wl,-z,now -Wl,--disable-new-dtags -Wl,--gc-sections -Wl,-rpath,/home/shommi2016/anaconda3/lib -Wl,-rpath-link,/home/shommi2016/anaconda3/lib -L/home/shommi2016/anaconda3/lib'
Using 'CPPFLAGS' from environment with value: '-DNDEBUG -D_FORTIFY_SOURCE=2 -O2 -isystem /home/shommi2016/anaconda3/include'
C compiler for the host machine: /home/shommi2016/anaconda3/bin/x86_64-conda_cos6-linux-gnu-cc (gcc 7.3.0 "x86_64-conda_cos6-linux-gnu-cc (crosstool-NG 1.23.0.449-a04d0) 7.3.0")
C linker for the host machine: /home/shommi2016/anaconda3/bin/x86_64-conda_cos6-linux-gnu-cc ld.bfd 1.23.0.444-4ea7
Using 'AR' from environment with value: '/home/shommi2016/anaconda3/bin/x86_64-conda_cos6-linux-gnu-ar'
Host machine cpu family: x86_64
Host machine cpu: x86_64
Found pkg-config: /usr/bin/pkg-config (0.29.1)
Run-time dependency libavformat found: YES 57.83.100
Run-time dependency libavcodec found: YES 57.107.100
Run-time dependency libavutil found: YES 55.78.100
Run-time dependency sdl2 found: YES 2.0.8
Configuring config.h using configuration
Program ./scripts/build-wrapper.sh found: YES
Build targets in project: 10

Found ninja-1.8.2 at /usr/bin/ninja
Using 'STRIP' from environment with value: '/home/shommi2016/anaconda3/bin/x86_64-conda_cos6-linux-gnu-strip'
Using 'STRIP' from environment with value: '/home/shommi2016/anaconda3/bin/x86_64-conda_cos6-linux-gnu-strip'
```

**Comment by rom1v** [comment]
> 1.8 because the prebuilt server .jar was that;

The prebuild server is provided for each scrcpy version. The latest: https://github.com/Genymobile/scrcpy/blob/master/BUILD.md#prebuilt-server

>     Run-time dependency libavformat found: YES 57.83.100
>     Run-time dependency libavcodec found: YES 57.107.100
>     Run-time dependency libavutil found: YES 55.78.100
>     Run-time dependency sdl2 found: YES 2.0.8

OK, so everything looks fine here.

> Using 'CC' from environment with value: '/home/shommi2016/anaconda3/bin/x86_64-conda_cos6-linux-gnu-cc'

What if you use gcc from your distrib repositories instead?

**Comment by sidommi** [comment]
Yes, now I changed to the latest prebuilt server (1.16), and with the same error.

I understand what you suggest to use gcc from distrib repositories. But I am not very good linux. Can you please tell me how to do it.

Thank you

**Comment by Biswa96** [comment]
> ../app/src/controller.h:5:10: fatal error: SDL2/SDL_mutex.h: No such file or directory

It is likely you are missing some packages.

**Comment by sidommi** [comment]
Resolved with instructions of @rom1v .
anaconda3 gcc repositories were set as path in the .bashrc. I removed them and also put a hard-stop within in .bashrc to stop looking at that path. like below 
```
PATH=$(echo "$PATH" | sed -e 's/:\/home\/shommi2016\/anaconda3$//')
```
So once the gcc path is set to default system repository everything went well. Probably I have to update the gcc within anaconda later.

Thanks a lot @rom1v. This app is wonderful and beats all the competition if any. I hope that support continues in the future.

Sid
